src/zenml/models/artifact_models.py

```diff
@@ -212,3 +218,16 @@ class ArtifactFilterModel(WorkspaceScopedFilterModel):
 
 class ArtifactRequestModel(ArtifactBaseModel, WorkspaceScopedRequestModel):
     """Request model for artifacts."""
+
+
+# ------ #
+# UPDATE #
+# ------ #
+
+
+class ArtifactUpdateModel(BaseModel):
+    """Artifact update model."""
+
+    name: Optional[str] = None
+    version: Optional[Union[str, int]] = None
```

Comment:
After redesigning some endpoints for MCP, I feel that `name` and `version` should not be here, but come from the call itself. They should be here only in case they can be updated, IMO. So, if you want to allow users to change name or version, it is fine.

Reply:
Yeah those are the attributes that users can change, not the identifier. Name needs to be here since we want to enable artifact renaming. However, as you correctly pointed out, version should probably be immutable 👍